contracts/EventBasedPredictionMarket.sol

```diff
@@ -0,0 +1,318 @@
+
+    /****************************************
+     *                EVENTS                *
+     ****************************************/
+
+    event TokensCreated(address indexed sponsor, uint256 indexed collateralUsed, uint256 indexed tokensMinted);
+    event TokensRedeemed(address indexed sponsor, uint256 indexed collateralReturned, uint256 indexed tokensRedeemed);
+    event PositionSettled(address indexed sponsor, uint256 collateralReturned, uint256 longTokens, uint256 shortTokens);
+
+    /****************************************
+     *               MODIFIERS              *
+     ****************************************/
+
+    modifier hasPrice() {
+        require(
+            getOptimisticOracle().hasPrice(address(this), priceIdentifier, expirationTimestamp, customAncillaryData)
+        );
+        _;
+    }
+
+    modifier requestInitialized() {
+        require(priceRequested, "Price not requested");
+        _;
+    }
+
+    /**
+     * @notice Construct the EventBasedPredictionMarket
+     * @param _pairName: Name of the long short pair tokens created for the prediction market.
+     * @param _priceIdentifier: Price identifier, registered in the DVM for the long short pair.
+     * @param _collateralToken: Collateral token used to back LSP synthetics.
+     * @param _customAncillaryData: Custom ancillary data to be passed along with the price request to the OO.
+     * @param _finder: DVM finder to find other UMA ecosystem contracts.
+     * @param _timerAddress: Timer used to synchronize contract time in testing. Set to 0x000... in production.
+     */
+    constructor(
+        string memory _pairName,
+        bytes32 _priceIdentifier,
+        ExpandedERC20 _collateralToken,
+        bytes memory _customAncillaryData,
+        FinderInterface _finder,
+        address _timerAddress
+    ) Testable(_timerAddress) {
+        expirationTimestamp = getCurrentTime(); // Set the request timestamp to the current block timestamp.
+
+        // Holding long tokens gives the owner exposure to the long position,
+        // i.e. the case where the answer to the prediction market question is YES.
+        longToken = new ExpandedERC20(string(abi.encodePacked(_pairName, " Long Token")), "PLT", 18);
+        // Holding short tokens gives the owner exposure to the short position,
+        // i.e. the case where the answer to the prediction market question is NO.
+        shortToken = new ExpandedERC20(string(abi.encodePacked(_pairName, " Short Token")), "PST", 18);
+
+        // Add burner and minter required roles to the long and short tokens.
+        longToken.addMinter(address(this));
+        shortToken.addMinter(address(this));
+        longToken.addBurner(address(this));
+        shortToken.addBurner(address(this));
+
+        finder = _finder;
+
+        collateralToken = _collateralToken;
+        customAncillaryData = _customAncillaryData;
+        priceIdentifier = _priceIdentifier;
```

Comment:
I didn't see any logic to make sure that the collateralToken and priceIdentifier are registered within UMA system. This is actually potentially dangerous, as you could deploy the contract and have people trade it, but then `initializeMarket()` would revert.

Example for how to do this: https://github.com/UMAprotocol/protocol/blob/master/packages/core/contracts/financial-templates/long-short-pair/LongShortPair.sol#L137

Additionally I'd be in favor of hardcoding `priceIdentifier` as `YES_OR_NO_QUERY` and not having the deployer need to set it.